crates/solver/src/settlement_access_list.rs

```diff
@@ -0,0 +1,104 @@
+use ethcontract::{Address, H256};
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    Client, Url,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize)]
+pub struct TenderlyRequest {
+    pub network_id: String,
+    pub block_number: u64,
+    pub from: Address,
+    pub input: String,
+    pub to: Address,
+    pub generate_access_list: bool,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct TenderlyResponse {
+    pub generated_access_list: Vec<AccessListItem>,
+}
+
+/// Had to introduce copy of the web3 AccessList because tenderly responds with snake_case fields
+/// and tenderly storage_keys field does not exist if empty (it should be empty Vec instead)
+#[derive(Debug, Deserialize)]
+pub struct AccessListItem {
+    /// Accessed address
+    pub address: Address,
+    /// Accessed storage keys
+    pub storage_keys: Option<Vec<H256>>,
+}
+
+impl From<AccessListItem> for web3::types::AccessListItem {
+    fn from(item: AccessListItem) -> Self {
+        Self {
+            address: item.address,
+            storage_keys: item.storage_keys.unwrap_or_default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TenderlyApi {
+    url: Url,
+    client: Client,
+    header: HeaderMap,
+}
+
+impl TenderlyApi {
+    #[allow(dead_code)]
+    pub fn new(url: Url, api_key: &str) -> Self {
+        Self {
+            url,
+            client: Client::new(),
+            header: {
+                let mut header = HeaderMap::new();
+                header.insert("x-access-key", HeaderValue::from_str(api_key).unwrap());
+                header
+            },
+        }
+    }
+}
+
+impl TenderlyApi {
+    #[allow(dead_code)]
+    pub async fn access_list(&self, body: TenderlyRequest) -> reqwest::Result<TenderlyResponse> {
```

Comment:
This can be moved into the existing `impl` block.

Reply:
Fixed.